updater/updater_test.go

```diff
@@ -0,0 +1,179 @@
+package updater
```

Comment:
This could use a test package, so ensure we only do blackbox testing.
```suggestion
package updater_test
```

Reply:
We access one not exported method, so I moved it back to whitebox testing. I made the mistake of having the example be part of the packge too, so I will revert that one later.